dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java

```diff
@@ -0,0 +1,273 @@
+import com.dotmarketing.portlets.templates.model.Template;
+import com.dotmarketing.util.PageMode;
+import com.dotmarketing.util.UUIDGenerator;
+import com.dotmarketing.util.WebKeys;
+import com.google.common.collect.ImmutableSet;
+import com.liferay.portal.model.User;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.mockito.stubbing.Answer;
+
+public class HTMLDiffUtilTest extends IntegrationTestBase {
+
+    private static ContentletAPI contentletAPI;
+    private static MultiTreeAPI multiTreeAPI;
+    private static String contentGenericId;
+    private static Host site;
+    private static Template template;
+    private static Folder folder;
+    private static User systemUser;
+    private static Container container;
+    private static Language defaultLang;
+    private static String uuid;
+
+    private static final String NOTHING_CHANGED = "Nothing Changed";
+
+    @BeforeClass
+    public static void prepare() throws Exception {
+
+        IntegrationTestInitService.getInstance().init();
+        LicenseTestUtil.getLicense();
+        contentletAPI = APILocator.getContentletAPI();
+        final LanguageAPI languageAPI = APILocator.getLanguageAPI();
+        multiTreeAPI = APILocator.getMultiTreeAPI();
+        systemUser = APILocator.systemUser();
+        site = new SiteDataGen().nextPersisted();
+        defaultLang = languageAPI.getDefaultLanguage();
+        final ContentTypeAPI contentTypeAPI = APILocator.getContentTypeAPI(systemUser);
+        final ContentType contentGenericType = contentTypeAPI.find("webPageContent");
+        contentGenericId = contentGenericType.id();
+
+        final String nameTitle = "anyTestContainer" + System.currentTimeMillis();
+        uuid = UUIDGenerator.generateUuid();
+        container = new ContainerDataGen()
+                .withContentType(contentGenericType, "$!{body}")
+                .friendlyName(nameTitle)
+                .title(nameTitle)
+                .nextPersisted();
+
+        PublishFactory.publishAsset(container, systemUser, false, false);
+        template = new TemplateDataGen().withContainer(container.getIdentifier(), uuid)
+                .nextPersisted();
+
+        folder = new FolderDataGen().site(site).nextPersisted();
+
+        PublishFactory.publishAsset(template, systemUser, false, false);
+    }
+
+    /**
+     * Given scenario: We have a page created out of a layout and a container. The Container holds a List of items.
+     * Expected Result:  We create a working copy and modify the list of items. The new items must replace the old ones.
+     * @throws Exception
+     */
+    @Test
+    public void Test_Page_With_Changes_Expect_Difference() throws Exception {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5511524)